service/resource/vaultcrt/create.go

```diff
@@ -0,0 +1,127 @@
+package vaultcrt
+
+import (
+	"context"
+
+	"github.com/giantswarm/certificatetpr"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/vaultcrt"
+	"github.com/giantswarm/vaultrole"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+
+	"github.com/giantswarm/cert-operator/service/key"
+)
+
+func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	currentSecret, err := toSecret(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredSecret, err := toSecret(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out if the secret has to be created")
+
+	var secretToCreate *apiv1.Secret
+	if currentSecret == nil {
```

Comment:
Nitpick: I almost always prefer short `if` branches. There is less information to carry when looking at the code. In this case it would be

```
if currentSecret != nil {
        r.logger.Log("cluster", key.ClusterID(customObject), "debug", "found out secret is already created")
        return nil, nil
}
```

Reply:
I agree with short branches but our definition of short seems to be different. The branch is 15 lines long. This is easily observable on almost every "normal" screen you work on. So this is still ok for me. Further you mean early returns which I have no strong feelings about at this point.